### PR TITLE
feat: credential setup and management (spec 002)

### PR DESCRIPTION
**Pull request by PetroczyP**
## Summary
- **4 new server modules**: `credential_store` (Fernet encryption, atomic writes), `credential_resolver` (Docker secret > env var > stored priority), `token_validator` (format + GitHub auth + Copilot SDK validation with verbose diagnostic errors), `web_routes` (setup wizard, status page, settings/rotation)
- **Web UI**: Server-rendered Jinja2 with monospace dark theme — setup wizard for first-time PAT onboarding, credential status page, token rotation
- **Security**: Tokens never in plaintext in logs/errors/repr, Fernet encryption with `chmod 600` key, `127.0.0.1` binding, content denylist preserved
- **Infrastructure**: Docker named volume for `/data/`, `cryptography` dependency added

## Test Plan
- [x] 524 tests pass, 95% coverage, 0 warnings
- [x] All 9 acceptance criteria verified (AC-1 through AC-9)
- [x] All 5 edge cases covered (key loss, credential invalidation, SDK unavailable, whitespace stripping)
- [x] Security tests: token masking in repr/logs/errors, no plaintext leakage
- [x] Docker build + compose validated (T026/T027)
- [x] PR review: 4 specialized agents (code, errors, tests, types) — all findings addressed

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Local web UI for credential setup, status and settings with masked token display, PRG flash messages, and in‑app rotation; multi-source resolution and server-side validation (format, GitHub probe, Copilot access).

* **Security**
  * Localhost-only binding, persistent data volume with owner-only permissions, encrypted on-disk credentials, secret-file support, strict masking and no-token-in-logs.

* **Documentation**
  * Full spec, contracts, plan, archival rounds, checklists and task/judge/status records added.

* **Tests**
  * Extensive test suites for resolver, store, validator, web routes and startup mappings.

* **Chores**
  * Added cryptography dependency and container/runtime adjustments for /data and secrets.

### CODE REVIEW

File: server/web_routes.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+            "can_rotate": can_rotate,
+            "message": None,
+            "error": None,
+        }
+
+        if source != CredentialSource.STORED:
+            # Externally managed — reject rotation
+            logger.warning("Rotation rejected: credential source is %s", source)
+            ctx["error"] = "Token is managed externally and cannot be rotated here."
+            return templates.TemplateResponse(request, "settings.html", ctx)
```
[the remark this answers is not in the document]
Comment (PetroczyP) [reply server/web_routes.py]: Fixed — added early return for `CredentialSource.NONE` with message "No token is configured. Please set one up first via /setup." before the externally-managed check.

File: tests/test_web_routes.py [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+
+from server.credential_resolver import CredentialSource, ResolvedCredential
+from server.token_validator import TokenValidationError
+
+
+def _create_test_app(resolver_result=None, resolver_source=CredentialSource.NONE):
+    """Create a FastAPI test app with mocked dependencies."""
+    from server.web_routes import create_router, mask_token
+    from server.main import app as real_app
+    from fastapi import FastAPI
```
[the remark this answers is not in the document]
Comment (PetroczyP) [reply tests/test_web_routes.py]: Fixed — removed unused `patch` import, unused `real_app` import, and `mask_token` from `_create_test_app` (each test that needs it imports locally).

File: server/token_validator.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+                raise TokenValidationError(
+                    _MSG_AUTH_LOW_CONFIDENCE, error_type="auth"
+                ) from None
+        except Exception as e:
+            logger.error("Unexpected error during Copilot validation: %s", e, exc_info=True)
+            raise TokenValidationError(
+                "An unexpected error occurred while validating Copilot access. "
+                "Check container logs for details and try again.",
+                error_type="sdk",
+            ) from e
```
[the remark this answers is not in the document]
Comment (PetroczyP) [reply server/token_validator.py]: Fixed — now logs only `type(e).__name__` (no exception string or traceback), and uses `from None` to suppress the exception chain. Prevents potential token leakage from HTTP library exceptions.
